shared/js/background/https.es6.js

```diff
@@ -95,6 +95,65 @@ class HTTPS {
         // If it falls to here, default to reqUrl
         return reqUrl
     }
+
+    /**
+     * Modify response headers on https pages to fix mixed content.
+     * 1. For main frame requests, add uprade-insecure-requests directive
+     * 2. For subrequests, edit Access-Control-Allow-Origin header if it exists
+     *    to allow resources to be loaded on the https version of site.
+     */
+    setUpgradeInsecureRequestHeader (request) {
+        // Skip header modifications if request is not https
+        if (request.url.indexOf('https://') !== 0) return {}
+
+        let headersChanged = false
+
+        if (request.type === 'main_frame') {
+            let cspHeaderExists = false
+
+            for (const header in request.responseHeaders) {
+                // If CSP header exists and doesn't include upgrade-insecure-request
+                // directive, append it.
+                if (request.responseHeaders[header].name.match(/Content-Security-Policy/i)) {
+                    cspHeaderExists = true
+                    const cspValue = request.responseHeaders[header].value
+                    // exit loop if upgrade-insecure-requests directive present
+                    if (cspValue.match(/upgrade-insecure-requests/i)) break
```

Comment:
I thought a bit about whether we should exit the loop early elsewhere once we've found the header we're interested in, but it seems some headers allow multiple entries (https://stackoverflow.com/a/4371395/). So this is probably good to leave as-is.